pkg/authorization/api/conversion_test.go

```diff
@@ -0,0 +1,424 @@
+			t.Fatal(err)
+		}
+		if !reflect.DeepEqual(rr, rr2) {
+			t.Errorf("rbac local data not preserved; the diff is %s", diff.ObjectDiff(rr, rr2))
+		}
+	}
+}
+
+func TestRBACClusterRoleBindingFidelity(t *testing.T) {
+	for i := 0; i < 100; i++ {
+		rcrb := &rbac.ClusterRoleBinding{}
+		rcrb2 := &rbac.ClusterRoleBinding{}
+		ocrb := &ClusterRoleBinding{}
+		fuzzer.Fuzz(rcrb)
+		if err := Convert_rbac_ClusterRoleBinding_To_api_ClusterRoleBinding(rcrb, ocrb, nil); err != nil {
+			t.Fatal(err)
+		}
+		if err := Convert_api_ClusterRoleBinding_To_rbac_ClusterRoleBinding(ocrb, rcrb2, nil); err != nil {
+			t.Fatal(err)
+		}
+		if !reflect.DeepEqual(rcrb, rcrb2) {
+			t.Errorf("rbac cluster binding data not preserved; the diff is %s", diff.ObjectDiff(rcrb, rcrb2))
+		}
+	}
+}
+
+func TestRBACRoleBindingFidelity(t *testing.T) {
+	for i := 0; i < 100; i++ {
+		rrb := &rbac.RoleBinding{}
+		rrb2 := &rbac.RoleBinding{}
+		orb := &RoleBinding{}
+		fuzzer.Fuzz(rrb)
+		if err := Convert_rbac_RoleBinding_To_api_RoleBinding(rrb, orb, nil); err != nil {
+			t.Fatal(err)
+		}
+		if err := Convert_api_RoleBinding_To_rbac_RoleBinding(orb, rrb2, nil); err != nil {
+			t.Fatal(err)
+		}
+		if !reflect.DeepEqual(rrb, rrb2) {
+			t.Errorf("rbac local binding data not preserved; the diff is %s", diff.ObjectDiff(rrb, rrb2))
+		}
+	}
+}
+
+func TestConversionErrors(t *testing.T) {
+	for _, test := range []struct {
+		name     string
+		expected string
+		f        func() error
+	}{
+		{
+			name:     "invalid origin role ref",
+			expected: `invalid origin role binding rolebindingname: attempts to reference role in namespace "ns1" instead of current namespace "ns0"`,
+			f: func() error {
+				return Convert_api_RoleBinding_To_rbac_RoleBinding(&RoleBinding{
+					ObjectMeta: api.ObjectMeta{Name: "rolebindingname", Namespace: "ns0"},
+					RoleRef:    api.ObjectReference{Namespace: "ns1"},
+				}, &rbac.RoleBinding{}, nil)
+			},
+		},
+		{
+			name:     "invalid origin subject kind",
+			expected: `invalid kind for origin subject: "fancyuser"`,
+			f: func() error {
+				return Convert_api_ClusterRoleBinding_To_rbac_ClusterRoleBinding(&ClusterRoleBinding{
+					Subjects: []api.ObjectReference{
+						{Kind: "fancyuser"},
+					},
+				}, &rbac.ClusterRoleBinding{}, nil)
+			},
+		},
+		{
+			name:     "invalid RBAC subject kind",
+			expected: `invalid kind for rbac subject: "evenfancieruser"`,
+			f: func() error {
+				return Convert_rbac_ClusterRoleBinding_To_api_ClusterRoleBinding(&rbac.ClusterRoleBinding{
+					Subjects: []rbac.Subject{
+						{Kind: "evenfancieruser"},
+					},
+				}, &ClusterRoleBinding{}, nil)
+			},
+		},
+	} {
+		if err := test.f(); err == nil || test.expected != err.Error() {
+			t.Errorf("%s failed: expected %q got %v", test.name, test.expected, err)
+		}
+	}
+}
+
+// rules with AttributeRestrictions should not be preserved during conversion
+func TestAttributeRestrictionsRuleLoss(t *testing.T) {
```

Comment:
Shouldn't this function be using a two way covers check?  Also, that probably needs checking come to think of it.